src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -269,6 +269,35 @@ public void WritePortBindingEnvironmentVariables(IResource resource)
         }
     }
 
+    internal void WriteDockerBuildArgs(DockerBuildArg[] buildArgs)
+    {
+        if (buildArgs is { Length: > 0 })
+        {
+            Writer.WriteStartObject("buildArgs");
+
+            for (var i = 0; i < buildArgs.Length; i++)
+            {
+                var buildArg = buildArgs[i];
+
+                var value = buildArg.Value;
+                if (string.IsNullOrEmpty(value))
+                {
+                    value = Environment.GetEnvironmentVariable(buildArg.Name);
```

Comment:
Should this instead come from the builder's `IConfiguration` rather than directly from the environment?

Reply:
That's a fair question, I'm not entirely sure.

The `ManifestPublishingContext` doesn't seem to have the builder's `IConfiguration` readily available.

Reply:
Why is it reading from the environment at all?

Reply:
User code can do that. It shouldn’t be done at this layer

Reply:
Where should this occur? When a value isn't provided, we'll use the current value that corresponds to the environment variable. One thing to note here, is that `docker build` has this logic where it provides values from the env var when they're not otherwise specified. So, perhaps this isn't needed at all?

> You can also use the `--build-arg` flag without a value, in which case the daemon propagates the value from the local environment into the Docker container it's building:

https://docs.docker.com/reference/cli/docker/image/build/#build-arg

Thoughts?

Reply:
Should the tool do this instead of it being hardcoded in the manifest?

Reply:
Yeah, I updated this to remove the reading of the env var altogether. Now it accepts name/value pair, with an `object` value, using similar logic as observed elsewhere. And it also allows for an empty value, which will let the underlying tooling get the env var.